src/algorithms/all_paths.c

```diff
@@ -92,10 +92,14 @@ size_t AllPaths
     size_t nodeCount = Graph_NodeCount(g);
     GrB_Matrix_new(&visited, GrB_BOOL, nodeCount, nodeCount);
 
-    size_t pathLen = maxLen-minLen;
+    unsigned int pathLen = (maxLen == PATH_UNBOUNDED) ? 16 : maxLen - minLen;
```

Comment:
(maxLen - minLen) can be quite large `[:*1..2^31]`
I think pathLen = min(16, maxLen - minLen) would be better.